enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,532 @@
+---
+title: infrastructure-external-platform-type
+authors:
+  - "@lobziik"
+  - "@elmiko"
+reviewers:
+  - "@dhellmann"
+  - "@mhrivnak"
+  - "@rvanderp3"
+  - "@mtulio"
+  - "@deads2k" 
+  - "@JoelSpeed"
+  - "@sinnykumari"
+  - "@danwinship"
+  - "@Miciah"
+  - "@openshift/openshift-team-windows-containers"
+approvers:
+  - "@dhellmann"
+  - "@deads2k"
+  - "@bparees"
+api-approvers:
+  - "@deads2k"
+  - "@JoelSpeed"
+creation-date: 2022-09-06
+last-updated: 2022-09-23
+tracking-link:
+  - https://issues.redhat.com/browse/OCPPLAN-9429
+  - https://issues.redhat.com/browse/OCPPLAN-8156
+see-also:
+  - "[KEP-2392 Cloud Controller Manager](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/2392-cloud-controller-manager)"
+  - "[OCP infrastructure provider onboarding guide](https://docs.providers.openshift.org/overview/)"
+  - "[Out-of-tree cloud provider integration support](https://github.com/openshift/enhancements/blob/master/enhancements/cloud-integration/out-of-tree-provider-support.md)"
+  - "[Platform Operators Proposal](https://github.com/openshift/enhancements/blob/master/enhancements/olm/platform-operators.md)"
+  - "[Capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)"
+---
+
+# Introduce new platform type "External" in the OpenShift specific Infrastructure resource
```

Comment:
Naming is hard.  Are we good with "External"?  Will it be clear what the platform is external to?  Does a word like pluggable, extension, auxiliary, ancillary, or supplemental fit better?

Reply:
"External" has been the name we have been using so far as it felt very similar to what was happening in upstream kubernetes around the in-tree -> out-of-tree cloud managers. i'm happy to explore other ideas but so far it seems that "External" has made sense to people. 

from the list you suggested, i think "pluggable" and "extension" make the most sense to me. we definitely want to denote that the "platform" is of a type that is not built in to openshift but that will provide some, if not all, of the behaviors that a user will expect from openshift.

what would your preferred name for this platform type be?

Reply:
I don't have any preference.  If "external" has precedence then that's a plus.  Just wanted to make sure we're good with it before we "commit" :).

Reply:
+1, and i appreciate the thoughtfulness =)

through all our discussions with the install flex working group, the wide cluster lifecycle openshift group, and the individual teams we have talked with, no one has pushed back strongly on the name. it seems to make sense what we are targeting, so i have been happy to continue going.

Reply:
I like this question a lot. It's no surprise that "external" makes sense when our perspective is that of openshift engineers. But will it make as much sense to the cloud providers who we want to use it? Great observation.

We use "external" to imply that the code, artifacts, management, and lifecycle of platform-specific components are handled external from openshift.

The value of this new platform is that it enables third-party platform components to be integrated at deployment time. Words like "custom", "plugin", "composite", and "extension" might do a better job of expressing that.

But I could also see us continuing with "external" to convey that platform components are owned by an external entity.